### PR TITLE
Add documentation for custom Docker networking using the CLI

### PR DESCRIPTION
**Pull request by timaa2k**

### CODE REVIEW

File: docs/source/dcos-docker-cli.rst [hunk 1]
```diff
@@ -54,6 +54,45 @@ To wait until DC/OS has finished installing, use the :ref:`the dcos-docker wait
...
+By default DC/OS clusters are launched on the ``docker0`` network.
+
+To launch a DC/OS cluster on a custom Docker network the network must first be created using the standard Docker CLI.
+During ``dcos-docker create`` the command line option ``--network`` then takes the name of the Docker network as a parameter.
+
+DC/OS nodes utilize an environment-specific ``ip-detect`` script to detect their current private IP address.
+The default ``ip-detect`` script used by ``dcos-docker`` does only account for the ``docker0`` network case.
+Therefore, in order for DC/OS to operate on a custom network a custom ``ip-detect`` script needs to be provided and put into the ``genconf`` directory before installing DC/OS.
+
+Luckily, the following IP detect script works for any custom Docker bridge network:
```
Comment (adamtheturtle) [inline docs/source/dcos-docker-cli.rst]: `S/Luckily, the/The`.

File: docs/source/dcos-docker-cli.rst [hunk 2]
```diff
@@ -54,6 +54,45 @@ To wait until DC/OS has finished installing, use the :ref:`the dcos-docker wait
 
 To use this cluster, it is useful to find details using the :ref:`the dcos-docker inspect <dcos-docker-inspect>` command.
 
+Using a custom Docker network
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default DC/OS clusters are launched on the ``docker0`` network.
```
Comment (adamtheturtle) [inline docs/source/dcos-docker-cli.rst]: ❓ `s/network/bridge network`.

File: docs/source/dcos-docker-cli.rst [hunk 3]
```diff
@@ -54,6 +54,45 @@ To wait until DC/OS has finished installing, use the :ref:`the dcos-docker wait
...
+
+    # Create ip-detect as mentioned above
+    $ docker network create custom-bridge
+    $ mkdir custom-genconf
+    $ mv ip-detect custom-genconf/ip-detect
+    $ dcos-docker create /path/to/dcos_generate_config.sh
+        --network custom-bridge
+        --genconf-dir ./custom-genconf
+
+Note that the custom Docker network is not cleaned up by the ``dcos-docker`` CLI.
```
Comment (adamtheturtle) [inline docs/source/dcos-docker-cli.rst]: ❓ Nit: `s/Note that the/The`.

File: docs/source/dcos-docker-cli.rst [hunk 4]
```diff
@@ -54,6 +54,45 @@ To wait until DC/OS has finished installing, use the :ref:`the dcos-docker wait
 
 To use this cluster, it is useful to find details using the :ref:`the dcos-docker inspect <dcos-docker-inspect>` command.
 
+Using a custom Docker network
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default DC/OS clusters are launched on the ``docker0`` network.
+
+To launch a DC/OS cluster on a custom Docker network the network must first be created using the standard Docker CLI.
+During ``dcos-docker create`` the command line option ``--network`` then takes the name of the Docker network as a parameter.
```
Comment (adamtheturtle) [inline docs/source/dcos-docker-cli.rst]: Use  `` :ref:`dcos-docker create` `` here and elsewhere for some magic linking.

**Comment by adamtheturtle** [comment]
Nice!

Please:
* Make the minor changes
* `make lint`
* Confirm the rendered docs look good to you
* Merge!

File: docs/source/dcos-docker-cli.rst (hunk 2 above)
Comment (timaa2k) [reply docs/source/dcos-docker-cli.rst]: Didn't address this one as I think it is not relevant here.

**Pull request merged**